src/main/java/de/tum/in/www1/artemis/repository/ExamRepository.java

```diff
@@ -49,6 +49,44 @@ public interface ExamRepository extends JpaRepository<Exam, Long> {
             """)
     List<Exam> findAllByEndDateGreaterThanEqual(@Param("date") ZonedDateTime date);
 
+    @Query("""
+            SELECT COUNT(exam)
+            FROM Exam exam
+            WHERE exam.course.testCourse = false
+                AND exam.endDate >= :#{#minDate}
+                AND exam.endDate <= :#{#maxDate}
+            """)
+    Integer countExamsWithEndDateGreaterThanEqualButLessOrEqualThan(@Param("minDate") ZonedDateTime minDate, @Param("maxDate") ZonedDateTime maxDate);
+
+    @Query("""
+            SELECT COUNT(exam)
```

Comment:
Shouldn’t this count students instead of exams? Same for line 81.

Reply:
I believe that it should not matter since a join is used (so one result per pair of exam and user is returned) and counting these is equal to the number of registered students in the matching exams, but I changed it to avoid confusion.